torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -612,6 +621,18 @@ RegisterOperators reg(
            return 0;
          },
          aliasAnalysisFromSchema()),
+     Operator(
+         "aten::backward(Tensor(a!) self, Tensor? gradient=None, bool? retain_graph=None, bool create_graph=False) -> ()",
+         [](Stack& stack) {
+           bool create_graph = pop(stack).toBool();
+           auto retain_graph = pop(stack).toOptional<bool>();
+           at::Tensor gradient = toOptionalTensor(pop(stack));
+           at::Tensor self = pop(stack).toTensor();
+           bool keep_graph = retain_graph ? retain_graph.value() : create_graph;
+           self.backward(gradient, keep_graph, create_graph);
+           return 0;
+         },
+         aliasAnalysisSpecialCase()),
```

Comment:
Just put aliasAnalysisConservative here and this will be correct.

Reply:
The alias_analaysis does not allow me to make it a conservative kind because of the assertion [here](https://github.com/pytorch/pytorch/blob/c6617b370b903b74a5647dd34e69fd0081713df8/torch/csrc/jit/passes/alias_analysis.cpp#L395). So what I did is that I made them the special_case kind and call `analyze_conservative` when handling these special cases, this happens before the assert so it should work properly.

Reply:
That assertion is bogus, just remove it.